src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -98,7 +99,13 @@ private static string FindDotnetFromEnvironmentVariable(string environmentVariab
                 string fullPathToDotnetFromRoot = Path.Combine(dotnet_root, exeName);
                 if (File.Exists(fullPathToDotnetFromRoot))
                 {
-                    return realpath(fullPathToDotnetFromRoot) ?? fullPathToDotnetFromRoot;
+                    if (!IsWindows)
+                    {
+                        fullPathToDotnetFromRoot = realpath(fullPathToDotnetFromRoot) ?? fullPathToDotnetFromRoot;
+                        return File.Exists(fullPathToDotnetFromRoot) ? dotnet_root : null;
```

Comment:
If `fullPathToDotnetFromRoot` was assigned the real path on the previous line, wouldn't we want to return the directory part of the new real path, rather than `dotnet_root` which contains the symlink?

Reply:
I agree. Fixed it in one more place: https://github.com/microsoft/MSBuildLocator/blob/20af00a8a63c4ffd366aeed035c7c1c960f183a8/src/MSBuildLocator/DotNetSdkLocationHelper.cs#L149